test/jit/test_freezing.py

```diff
@@ -1939,7 +1950,42 @@ def test_layernorm(self):
 
                 for layernorm in (torch.nn.LayerNorm(param[1]),
                                   torch.nn.LayerNorm(param[1], elementwise_affine=False)):
-                    sub_model = torch.nn.Sequential(param[0], layernorm)
+                    # to generate non inplace tests we extend the use of layernorm's input
+                    for inplace in (True, False):
+                        sub_model = torch.nn.Sequential(param[0], layernorm) if inplace else ResidualLayernorm(param[0], layernorm)
+                        sub_model.eval()
+                        mod = torch.jit.freeze(torch.jit.script(sub_model))
+                        self.run_pass("convert_frozen_ops_to_mkldnn", mod.graph)
+                        # if weight and bias are present and shape is the last dimension
+                        # we should convert `aten::layer_norm` to `prim::MKLDNNLayerNorm`
+                        if layernorm.elementwise_affine and len(param[1]) == 1:
+                            inplace_suffix = "_" if inplace else ""
+                            FileCheck().check("prim::MKLDNNLayerNorm" + inplace_suffix).check_count("aten::to_dense", 1, exactly=True).run(mod.graph)
+                        else:
+                            FileCheck().check_count("aten::to_dense", 1, exactly=True).run(mod.graph)
+                        self.assertTrue(torch.allclose(sub_model(param[2]), mod(param[2]), 1e-04, 1e-04))
+
+    @skipIfNoTorchVision
+    def test_layernorm2(self):
+        with set_default_dtype(torch.float):
+            model = torchvision.models.resnet18()
+
+
+            class ResidualLayernorm(torch.nn.Module):
```

Comment:
is this needed if we're testing layernorm inplace above ?

Reply:
I think so. I added `inplace` for mostly documentation purposes. We can't pass it directly to MKLDNN. We use it to create different graphs where we can use inplace (`Sequential`) and we cannot ( `ResidualLayerNorm`) since the input to the layernorm is also used *after* the layernorm so we can't inplace.

Reply:
Could `F.layer_norm_`, `F.layer_norm(inplace = True)`, `nn.LayerNorm(inplace = True)` explicit variants be supported as well?